party_time/party_time.lua

```diff
@@ -0,0 +1,345 @@
+local os = require("os")
+local ui = require("ui")
+local string = require("string")
+local command = require("command")
+local packets = require("packets")
+local settings = require("settings")
+local treasure = require("treasure")
+
+local pt_settings = {
+    ui = {
+        x = 145,
+        y = 440,
+        enabled = true,
+    },
+    auto = {
+        accept_invites = true,   -- also for auto sending invites upon party requests
+        decline_invites = false, -- also ignores party requests from players on blacklists
+    },
+    blacklist = {}, 
+    whitelist = {},
+}
+
+pt_settings = settings.load(pt_settings, 'settings.lua')
+
+local invite_dialog = {}
+local invite_dialog_state = {
+    title = 'Party Invite',
+    style = 'normal',
+    x = pt_settings.ui.x,
+    y = pt_settings.ui.y,
+    width = 179,
+    height = 96,
+    resizable = false,
+    moveable = true,
+    closable = true,
+}
+local unhandled_requests = {}
+local unhandeled_invite = false
+
+
+local alias = {
+    add = {
+        ['add'] = true,
+        ['a'] = true,
+        ['+'] = true,
+    },
+    remove = {
+        ['remove'] = true,
+        ['rm'] = true,
+        ['r'] = true,
+        ['-'] = true,
+    },
+    on = {
+        ['true'] = true,
+        ['yes'] = true,
+        ['on'] = true,
+        ['y'] = true,
+    },
+    off = {
+        ['false'] = true,
+        ['off'] = true,
+        ['no'] = true,
+        ['n'] = true,
+    }
+}
+
+-- Command Arg Types
+-- Create and Register types
+command.arg.register('names', '<names:string(%a+)>*')
+
+command.arg.register_type('boolean', {
+    check = function(str)
+        if alias.on[str] then
+            return true
+        elseif alias.off[str] then
+            return false
+        end
+
+        error('Expected a boolean value.')
+    end
+})
+
+-- Addon Command Handlers
+-- Seven main commands additoinal sub_commands within
+local pt_command = command.new("pt")
+
+function join(name)
+    command.input("/prcmd add " .. name)
+end
+
+pt_command:register("j", join, '<name:string>')
+pt_command:register("join", join, '<name:string>')
+
+
+function invite(...)
+    for _, name in pairs({...}) do
+        command.input("/pcmd add " .. name)
+        name = name:gsub("^%l", string.upper)
+        if unhandled_requests[name] then
+            unhandled_requests[name] = nil
+        end
+    end
+end
+
+pt_command:register("i", invite, '{names}')
+pt_command:register("invite", invite, '{names}')
+
+
+function blacklist(section, sub_cmd, ...)
+    local names = {...}
+    if alias.add[sub_cmd] then
+        for _, name in pairs(names) do
+            pt_settings.blacklist[section][name] = true
+        end
+    elseif alias.remove[sub_cmd] then
+        for _, name in pairs(names) do
+            pt_settings.blacklist[section][name] = nil
+        end
+    end
```

Comment:
I may be biased seeing how I wrote the lib and how sets are the best thing ever, but have you considered using sets? :O

Would significantly simplify the code (assuming you also used sets to define the settings, as I mentioned before:
```lua
    local names = set({...})
    if alias.add[sub_cmd] then
        pt_settings.blacklist[section] = pt_settings.blacklist[section] + names
    else
        pt_settings.blacklist[section] = pt_settings.blacklist[section] - names
    end
```

Reply:
Getting errors using `+` or `:union()`

Reply:
Addressed in 6e27c21